File: benchmarks/string/regex_100k.js
const lines = Array.from({length: 100000}, (_, i) => `The quick brown fox jumps over the lazy dog #${i}`);
const pattern = /\b\w{5,}\b/g;
const start = Date.now();
let count = 0;
for (const line of lines) { count += (line.match(pattern) || []).length; }
console.log((Date.now() - start) * 1000);

File: benchmarks/string/regex_100k.go
package main

import (
	"fmt"
	"regexp"
	"time"
)

func main() {
	lines := make([]string, 100000)
	for i := range lines {
		lines[i] = fmt.Sprintf("The quick brown fox jumps over the lazy dog #%d", i)
	}
	pattern := regexp.MustCompile(`\b\w{5,}\b`)
	start := time.Now()
	count := 0
	for _, line := range lines {
		count += len(pattern.FindAllString(line, -1))
	}
	fmt.Println(time.Since(start).Microseconds())
}
